subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskCachingHashesLogger.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.execution;
+
+import com.google.common.hash.HashCode;
+import org.gradle.api.Task;
+import org.gradle.caching.BuildCacheKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class TaskCachingHashesLogger implements TaskCachingHashesListener {
+    private final Logger logger = LoggerFactory.getLogger(SkipCachedTaskExecuter.class);
+
+    @Override
+    public void inputsCollected(Task task, BuildCacheKey key, TaskCachingInputs hashes) {
+        logger.info("Cache key for {} is {}", task, key);
+        logger.info("{} classloader hash: {}", task, hashToStringRepresentation(hashes.getClassLoaderHash()));
```

Comment:
I am not very passionate about that. I chose info since we also are going to publish the hashes to cloud services - so info seemed adequate. We are already publishing all the single hashes in [DefaultBuildCacheKeyBuilder](https://github.com/gradle/gradle/blob/0eeac61ab99137123cf3c8bc2f6e0ab407c039ce/subprojects/core/src/main/java/org/gradle/caching/internal/DefaultBuildCacheKeyBuilder.java#L137-L137) on the debug level. This is another reason why I chose info level here. WDYT?